compiler/optimizer/OMROptimizationManager.cpp

```diff
@@ -49,7 +49,7 @@ OMR::OptimizationManager::OptimizationManager(TR::Optimizer *o, OptimizationFact
         _numPassesCompleted(0),
         _optData(NULL),
         _optPolicy(NULL),
-        _enabled(false),
+        _enabled(!o->comp()->isDisabled(optNum)),
```

Comment:
It can be better explained with the following use case: Opt A will do X if opt B is enabled. Currently if opt B is not even supported by a project, !comp()->isDisabled() will still return true, which is wrong. With this PR, if an opt is not supported by a project, we will not create an optimization manager for that opt so its _enabled field is nonexistent. Instead we will call optimizer->isEnabled() which in turn checks to see if an opt is supported first and if it's not, it will return false; otherwise it will return the value of the opt's _enabled field.

A: Yes, an opt manager is created even if the opt has been disabled (but is supported), however the opt class it manages will not be created. The opt manager class is a very lightweight class meant to hold "static" information about an opt such as its name, opts in its group if it's a group of opts, flags it needs to set, project-specific heuristics for the opt (which was added later), etc. The actual opt class is the one holding any data structures it needs during a particular pass of an opt. This can take a lot of memory depending on the opt and it used to live for the duration of the optimizer whether it was disabled or not.